### DISCUSSION TITLE
Add a background task system, like ActiveJob in Rails

### DISCUSSION THREAD

**Discussion by pschols** [martenframework/marten#125] [Ideas]
Given Marten's 'batteries included' philosophy, it would be great to have a background job system in Marten that works with all databases supported by Marten.

**Comment by treagod** [comment]
I know it's not directly part of Marten, but @ellmetha has created a post how to use [Mosquito to execute background jobs](https://dev.to/ellmetha/background-job-processing-with-marten-and-mosquito-2i3o)

**Comment by pschols** [comment]
Thanks, I had read the blog post and that's indeed a good way to get started. 
Over time, it would be nice to have this integrated into Marten, allowing to use the database that's being used for the rest of the project.
